ledger/shelley.go

```diff
@@ -441,6 +441,40 @@ type ShelleyProtocolParameterUpdate struct {
 	MinUtxoValue uint `cbor:"15,keyasint"`
 }
 
+const (
+	NonceType0 = 0
+	NonceType1 = 1
+)
+
+type Nonce struct {
+	cbor.StructAsArray
+	Type  uint `cbor:"0,keyasint"`
+	Value any  `cbor:"1,keyasint,omitempty"`
```

Comment:
The nonce value is `bytes .size 32` according to the CDDL, so we should use `[32]byte` instead of `any`

Reply:
When I set to [32]byte, it cannot unmarshal positive integer into Go struct field struct { StructAsArray cbor.StructAsArray; Type uint; Value [32]uint8 }.Value of type [32]uint8

Reply:
Then I tried to use temp struct

```
		tmp := struct {
			cbor.StructAsArray
			Type  uint
			Value []byte
		}{}

```
And got
cbor: cannot unmarshal positive integer into Go struct field struct { StructAsArray cbor.StructAsArray; Type uint; Value []uint8 }.Value of type []uint8

Reply:
I tried also

```
		tmp := struct {
			cbor.StructAsArray
			Type  uint
			Value []uint8
		}{}
```
I am not sure what does error want to tell me